manifests/package.pp

```diff
@@ -144,20 +160,32 @@
 
   }
 
-  package { $pkg_title:
-    ensure   => $pkg_version,
-    name     => $pkg_name,
-    source   => $pkg_source,
-    require  => $pkg_require,
-    provider => $pkg_provider,
+  case $::osfamily {
+    'Darwin': {
+      package { $pkg_title:
+        ensure   => present,
```

Comment:
instead of duplicating code here you could use the `?` operator

```puppet
$package_ensure = $::osfamily ? {
  default  => $pkg_version,
  'Darwin' => 'present',
}
```

then use `$package_ensure` as the value for the `ensure` attribute.

Reply:
I changed it, but pkgdmg provider doesn't have upgradeable feature, so I changed it back. Here is the error I had

```
==> macos-client: Error: Parameter ensure failed on Package[sensu]: Provider pkgdmg must have features 'upgradeable' to set 'ensure' to 'latest' at /etc/puppetlabs/code/environments/production/modules/sensu/manifests/package.pp:167
```